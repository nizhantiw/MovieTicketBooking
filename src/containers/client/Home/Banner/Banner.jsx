import React, { Component } from "react";
import "bootstrap/dist/css/bootstrap.min.css";
import c from "./carousal/c.jpg"
import b from "./carousal/b.jpg"
import aa from "./carousal/aa.jpg"
import {
  Carousel,
  CarouselItem,
  CarouselControl,
} from "reactstrap";

const items = [
  {
    src: c,
    altText: "Slide 1",
    caption: "Slide 1"
  },
  {
    src:b,
     altText: "Slide 2",
    caption: "Slide 2"
  },
  {
    src:aa,
     altText: "Slide 3",
    caption: "Slide 3"
  }
];

var lastX = 0;
var currentX = 0;

class Banner extends Component {
  constructor(props) {
    super(props);
    this.state = { activeIndex: 0 };
    this.next = this.next.bind(this);
    this.previous = this.previous.bind(this);
    this.goToIndex = this.goToIndex.bind(this);
    this.onExiting = this.onExiting.bind(this);
    this.onExited = this.onExited.bind(this);
  }

  onExiting() {
    this.animating = false;
  }

  onExited() {
    this.animating = false;
  }

  next() {
    if (this.animating) return;
    const nextIndex =
      this.state.activeIndex === items.length - 1
        ? 0
        : this.state.activeIndex + 1;
    this.setState({ activeIndex: nextIndex });
  }

  previous() {
    if (this.animating) return;
    const nextIndex =
      this.state.activeIndex === 0
        ? items.length - 1
        : this.state.activeIndex - 1;
    this.setState({ activeIndex: nextIndex });
  }

  goToIndex(newIndex) {
    if (this.animating) return;
    this.setState({ activeIndex: newIndex });
  }

  handleMouse = e => {
    e.persist();
    let type = e.type.toLowerCase();
    if (type === "dragstart") {
      lastX = e.clientX;
    } else {
      if (lastX === 0 || e.clientX === 0 || lastX === e.clientX) {
        return;
      }
      if (e.clientX > lastX) {
        this.previous();
        console.log("swife right");
      } else {
        this.next();
        console.log("swife left");
      }
    }
  };

  handleTouch = e => {
    e.persist();
    let type = e.type.toLowerCase();

    if (type === "touchstart") {
      lastX = e.touches[0].clientX;
    }

    if (type === "touchmove") {
      currentX = e.touches[0].clientX;
    }

    if (type === "touchend") {
      if (lastX === 0 || currentX === 0 || lastX === currentX) {
        return;
      }
      if (currentX > lastX) {
        this.previous();
        console.log("swife right");
      } else {
        this.next();
        console.log("swife left");
      }
    }
  };

  render() {
    const { activeIndex } = this.state;

    const slides = items.map(item => {
      return (
        <CarouselItem
          onExiting={this.onExiting}
          onExited={this.onExited}
          key={item.src}
        >
          <img
            style={{ width: "100%", height: "50%"}}
            src={item.src}
            alt={item.altText}
            onTouchStart={e => this.handleTouch(e)}
            onTouchMove={e => this.handleTouch(e)}
            onTouchEnd={e => this.handleTouch(e)}
            onDragStart={e => this.handleMouse(e)}
            onDragEnd={e => this.handleMouse(e)}
          />
        </CarouselItem>
      );
    });

    return (
      <div class="carousel">
      <Carousel 
        activeIndex={activeIndex}
        next={this.next}
        previous={this.previous}
        interval={false}
      >
        {slides}
        <CarouselControl id="CarouselControl1"
          direction="prev"
          directionText="Previous"
          onClickHandler={this.previous}
        />
        <CarouselControl id="CarouselControl2"
          direction="next"
          directionText="Next"
          onClickHandler={this.next}
        />
      </Carousel>
      </div>
    );
  }
}

export default Banner;
