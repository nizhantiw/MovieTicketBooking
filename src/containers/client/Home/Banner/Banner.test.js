import Banner from './Banner';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow, configure } from 'enzyme';
import renderer from 'react-test-renderer'
import { render, fireEvent } from "@testing-library/react";

configure({adapter: new Adapter()});

it("renders correctly", () => {
  const wrapper = shallow(
    <Banner />
  );
  expect(wrapper).toMatchSnapshot();
});


it('renders a snapshot', () => {
  const tree = renderer.create(<Banner/>).toJSON();
  expect(tree).toMatchSnapshot();
});



   it('renders results after search', () => {
    const wrapper = shallow(<Banner/>);
    expect(wrapper.find('Carousel').length).toEqual(1);
  });

  it('renders results after search', () => {
    const wrapper = shallow(<Banner/>);
    expect(wrapper.find('img').length).toEqual(3);
  });

  it('renders results after search', () => {
  const wrapper = shallow(<Banner />);
  const button = wrapper.find('Carousel');
  expect(button).toHaveLength(1);
});

it('renders results after search', () => {
  const wrapper = shallow(<Banner />);
  const button = wrapper.find('CarouselControl');
  expect(button).toHaveLength(2);
});

it('renders results after search', () => {
  const wrapper = shallow(<Banner />);
  const button = wrapper.find('#CarouselControl1');
  expect(button).toHaveLength(2);
  button.simulate('click');
});


it("show the sended message", () => {
  const onSubmit = jest.fn();
  const { getByLabelText, getByText } = render(<Banner />);
  const inputValue = "abc";
  const inputValue2 = "abc";


  fireEvent.change(getByLabelText("Previous"), { target: { value: inputValue } });
  fireEvent.change(getByLabelText("Next"), { target: { value: inputValue2 } });
  fireEvent.click(getByText(/Previous/i));
  fireEvent.click(getByText(/Next/i));

  //expect(onSubmit).toBeCalled();
  expect(onClickHandler).toHaveBeenCalledTimes(2);
});
