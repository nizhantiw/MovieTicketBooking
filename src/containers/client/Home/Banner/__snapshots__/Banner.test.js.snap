// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  class="carousel"
>
  <div
    className="carousel slide"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="carousel-inner"
    >
      <div
        className="carousel-item active"
      >
        <img
          alt="Slide 1"
          onDragEnd={[Function]}
          onDragStart={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          src="/carousal/c.jpg"
          style={
            Object {
              "height": "50%",
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="carousel-item"
      >
        <img
          alt="Slide 2"
          onDragEnd={[Function]}
          onDragStart={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          src="/carousal/b.jpg"
          style={
            Object {
              "height": "50%",
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="carousel-item"
      >
        <img
          alt="Slide 3"
          onDragEnd={[Function]}
          onDragStart={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          src="/carousal/aa.jpg"
          style={
            Object {
              "height": "50%",
              "width": "100%",
            }
          }
        />
      </div>
    </div>
    <label
      htmlFor="Previous"
    >
      Previous
    </label>
    <a
      className="carousel-control-prev"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
        }
      }
      tabIndex="0"
    >
      <span
        aria-hidden="true"
        className="carousel-control-prev-icon"
      />
      <span
        className="visually-hidden"
      >
        Previous
      </span>
    </a>
  </div>
</div>
`;

exports[`renders correctly 1`] = `ShallowWrapper {}`;
